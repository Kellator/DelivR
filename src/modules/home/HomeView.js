import React from 'react';
import SearchPage from './searchPage';
import '../../grid.css';
import '../../index.css';


const HomeView = () => ( 
    <div className="Grid Grid-column search-div">
        <div className="Grid-cell">
            <div className="Grid Grid-column ">
                <h1 className="title Grid-cell Grid-cell--center title"> DelivR </h1>   
                <p className=" home-text Grid-cell Grid-cell--center"> Hungry?  Nothing in the kitchen?  Don't want to go out? </p>
                <p className="home-text Grid-cell Grid-cell--center ">Let DelivR find you dinner that delivers!</p> 
            </div>
        </div>
        <div className="Grid-cell">
            <SearchPage />
        </div>
    </div>
);
{/*const mapStateToProps = (state, props) => ({
	search: {
		cuisineSearch: props.cuisineSearch,
		locationSearch: props.locationSearch
	}
});*/}
export default HomeView;