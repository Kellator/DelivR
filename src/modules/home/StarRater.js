import React from 'react';
import '../../grid.css';
import '../../index.css';

function StarRater(props) {
    const stars = [];
    for (let i=0; i<5; i++) {
        let className;
        if (i < props.rating) {
            className = 'fa fa-star';
        }
        else {
            className = 'fa fa-star-o';
        }
        const star = (
            <i className={className} key={i}></i>
        );
        stars.push(star);
    }

    return (
        <div className="result-star result-text">
            <span className="star-rater result-text">
                {stars}
            </span>
        </div>
    );
}
StarRater.defaultProps = {
    rating: 0
};

export default StarRater;