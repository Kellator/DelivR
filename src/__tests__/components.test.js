import React from 'react';
import StarRater from '../modules/home/StarRater';
import NoResult from '../modules/home/noResult';
import Result from '../modules/home/result';
import HomeView from '../modules/home/HomeView';
import SearchPage from '../modules/home/SearchPage';
import ReturnedResultList from '../modules/home/vis-result-list';
import renderer from 'react-test-renderer';
import { NavLink } from 'react-router-dom';

test('StarRater component render without throwing error', () => {
	const component = renderer.create(
		<StarRater />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

test('NoResult component renders without throwing error', () => {
	const component = renderer.create(
		<NoResult />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

test('Result component renders without throwing error', () => {
	const component = renderer.create(
		<Result img="image"/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

test('HomeView component renders without throwing error', () => {
	const component = renderer.create(
		<HomeView />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})

test('ReturnedResultList component renders without throwing error', () => {
	const component = renderer.create(
		<ReturnedResultList />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})