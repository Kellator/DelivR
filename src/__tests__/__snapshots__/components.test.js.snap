// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRater component render without throwing error 1`] = `
<div
  className="result_star"
>
  <span
    className="center-text gray"
  >
    <i
      className="fa fa-star-o"
    />
    <i
      className="fa fa-star-o"
    />
    <i
      className="fa fa-star-o"
    />
    <i
      className="fa fa-star-o"
    />
    <i
      className="fa fa-star-o"
    />
  </span>
</div>
`;
